expo/expo
# chore(cli): Bump urql and related packages (and replace graphql-tag)

# Why

These dependencies haven't been bumped in a while, and I've seen prior issues from people where our dependency on `graphql` causes them dependency conflicts. This is generally more likely to happen with `graphql` due to how it only likes to have one instance of itself present for server-side logic (an issue they _might_ resolve for v17, or not)

Bumping `@urql/core` (and `@urql/exchange-retry`) basically without many internal changes gets us a replacement for `graphql-tag` and removes the dependency on `graphql`.

# How

- Replace usage of `graphql-tag` with `gql` from `@urql/core`
- Update `@urql/core` client types
- Update `packages/@expo/cli/src/api/graphql/queries/AppQuery.ts` to interpolate the fragment directly
- Replace import of `GraphQLError` from `graphql` with `@0no-co/graphql.web` since we removed our dependency on `graphql` and it's used for `instanceof` checks

# Test Plan

- Run `./build/bin/cli whoami` in `@expo/cli`
- Run `./build/bin/cli login` in `@expo/cli` and follow login steps
- Add `debugExchange` from `@urql/core` to client and run `./build/bin/cli`

I believe this should check most of the existing GraphQL pipeline.

# Checklist
